src/include/migraphx/op/pooling.hpp

```diff
@@ -40,29 +40,27 @@ struct pooling
 
     shape compute_shape(std::vector<shape> inputs) const
     {
-        check_shapes{inputs, *this}.has(1).only_dims(4);
+        check_shapes{inputs, *this}.has(1);
 
         const shape& input = inputs.at(0);
         auto t             = input.type();
 
-        assert(lengths[0] <= (input.lens()[2] + 2 * padding[0]));
-        assert(lengths[1] <= (input.lens()[3] + 2 * padding[1]));
+        auto input_lens = input.lens();
+        size_t kdims    = input_lens.size() - 2;
 
```

Comment:
We need a bunch of checks to ensure the dimensions of the inputs, padding, strides, lengths to match since it can be any numbers independently.